docs/e2e/extension-e2e-guidelines.md

```diff
@@ -314,3 +315,50 @@ Test Name: Connects to a Hardware wallet for Trezor
 ```
 
 ✅ Proposed solution: The Trezor import flow involves opening the Trezor website, then the user takes additional steps on that website to connect the device. A fake version of this website could be created for testing, and the test build could be updated to use this fake version. It's also worth investigating a phishing detection solution, such as replacing Github.com with an empty page.
+
+## Page Object Model (POM)
+
+POM, or Page Object Model, is a design pattern commonly utilized for automating test cases.
+
+The Page Object is an object-oriented class that acts as an interface for the page of the application under test. Page classes contain web elements and methods to interact with these elements. While automating test cases, we create objects of these Page Classes and interact with web elements by calling the methods of these classes.
+
+### Composition of Each Class Page
+
+Each class page is composed of:
+
+- **Selectors**: HTML elements.
+- **Action Methods**: Methods for interacting with the elements.
+- **Check Methods**: Methods that assert the status of elements.
+
+The testcase spec creates the necessary page class objects and utilizes their methods to perform actions.
+
+### Best Practices
+
+- A Page Object should represent meaningful elements of a page and not necessarily a complete page. It can also be a small component, like a navbar.
+- All selectors should be kept in the page class file. Carefully define selectors for elements in page classes, opting for robust selectors since they will be extensively used across various locations.
+- The page class should contain properties and methods, or be composed of objects that expose access.
+- Page Objects should remain independent and not invoke other Page Objects to prevent circular references, ensuring they are typically isolated from each other. For handling complex workflows that require interaction across multiple pages, "Processes" should be implemented. This approach enables the incorporation of all relevant Page Objects to support specific flows, such as sending a transaction or creating a swap. A dedicated "Processes" folder is used to organize and manage these complex workflows.
```

Comment:
Again, de-capitalizing "page object":

``` suggestion
- Page objects should remain independent and not invoke other page objects to prevent circular references, ensuring they are typically isolated from each other. For handling complex workflows that require interaction across multiple pages, "Processes" should be implemented. This approach enables the incorporation of all relevant page objects to support specific flows, such as sending a transaction or creating a swap. A dedicated `Processes` folder is used to organize and manage these complex workflows.
```

De-capitalizing "processes" as well:

``` suggestion
- Page objects should remain independent and not invoke other page objects to prevent circular references, ensuring they are typically isolated from each other. For handling complex workflows that require interaction across multiple pages, _processes_ should be implemented. This approach enables the incorporation of all relevant page objects to support specific flows, such as sending a transaction or creating a swap. A dedicated `processes` folder is used to organize and manage these complex workflows.
```